commit e3ee697515c1c8fa1480f03ed24e0265380fb323
Author:     Kalpak Shah <kalpak@clusterfs.com>
AuthorDate: Thu May 22 16:38:19 2008 +0000

Print dir inode number when displaying corrupt hash code warning

Bugzilla-ID: b=9570
Signed-off-by: Kalpak Shah <kalpak@clusterfs.com>
Reviewed-by: Andreas Dilger <adilger@clusterfs.com>
Reviewed-by: Girish Shilamkar <girish@clusterfs.com>

Index: linux-stage/fs/ext4/namei.c
===================================================================
--- linux-stage.orig/fs/ext4/namei.c
+++ linux-stage/fs/ext4/namei.c
@@ -371,6 +371,6 @@ dx_probe(const struct qstr *d_name, stru
 	if (root->info.hash_version != DX_HASH_TEA &&
 	    root->info.hash_version != DX_HASH_HALF_MD4 &&
 	    root->info.hash_version != DX_HASH_LEGACY) {
-		ext4_warning_inode(dir, "Unrecognised inode hash code %u",
-				   root->info.hash_version);
+		ext4_warning_inode(dir, "Unrecognised inode hash code %u for directory "
+				   "%lu", root->info.hash_version, dir->i_ino);
 		goto fail;
